### PR TITLE
addition of ARMv7 and ARM64 builds

### PR DESCRIPTION
**Pull request by dave93cab**
Removes:
- istio-linux

Adds:
- istio-linux-amd64 (replaces istio-linux)
- istio-linux-armv7
- istio-linust-arm64

Follow-up:
Remove istio-linux from istio/istio

Depends-On: https://github.com/istio/istio/pull/22381

### CODE REVIEW

**Comment by sdake** [comment]
@dave93cab note I changed the Depends-On - I think you had it in reverse :) While Depends-On doesn't currently do anything in our T&R infrastructure, I am hopeful @clarketm can consider a way to enforce merge ordering with this field.

Cheers
-steve

File: pkg/build/archive.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ func Archive(manifest model.Manifest) error {
 	}
 
 	// We build archives for each arch. These contain the same thing except arch specific istioctl
-	for _, arch := range []string{"linux", "osx", "win"} {
+	for _, arch := range []string{"linux-amd64", "linux-armv7", "linux-arm64", "osx", "win"} {
```
Comment (sdake) [inline pkg/build/archive.go]: I believe you need to retain `linux` here, then we can merge this PR.

Next, we would merge PR: https://github.com/istio/istio/pull/22381

A follow-on PR would then remove `linux` from the string slice.

**Comment by sdake** [comment]
> @dave93cab note I changed the Depends-On - I think you had it in reverse :) While Depends-On doesn't currently do anything in our T&R infrastructure, I am hopeful @clarketm can consider a way to enforce merge ordering with this field.
> 
> Cheers
> -steve

On further review of my past work and this PR, you had the Depends-On field right. I restored the Depends-On field to the comment.

Cheers
-steve

**Comment by dave93cab** [comment]
Thanks for the review @sdake

**Comment by dave93cab** [comment]
@howardjohn @sdake 

I fixed up the validation which caused first run to fail (was looking for linux folder not linux-amd64),
the test failure now looks unrelated to this change:

` 2020-03-28T16:21:14.628007Z	info	Check failed: check Grafana failed: dashboards out of sync, release contains map[istio-mesh-dashboard:{} istio-performance-dashboard:{} istio-service-dashboard:{} istio-workload-dashboard:{} mixer-dashboard:{} pilot-dashboard:{}], manifest contains map[citadel-dashboard:{} galley-dashboard:{} istio-mesh-dashboard:{} istio-performance-dashboard:{} istio-service-dashboard:{} istio-workload-dashboard:{} mixer-dashboard:{} pilot-dashboard:{}]
Error: release validation FAILED `

**Comment by howardjohn** [comment]
My bad, I broke that from an unrelated change. I'll fix that in another PR
(or you can remove the galley and citadel dashboard expectation from your
PR if you want)

**Comment by sdake** [comment]
@dave93cab You are highly capable of complex technical work and it would be good to welcome you aboard as a member. The main benefit of membership is that a maintainer does not need to run `/ok-to-test` on your PRs, speeding up the PR review process.

The process is here if you wish to proceed: https://github.com/istio/community/blob/master/ROLES.md#member

**Comment by howardjohn** [comment]
https://github.com/istio/release-builder/pull/181 will fix the tests

**Comment by dave93cab** [comment]
@howardjohn @ericvn @sdake tests passed

**Comment by BlackMix** [comment]
@dave93cab thanks for add arm64, I just need run istio in arm64 and this will help me.

**Pull request merged**